src/systems/sprites.js

```diff
@@ -187,224 +177,234 @@ function createGeometry(maxSprites) {
     indices[i * 3 * 2 + 4] = i * 4 + 2;
     indices[i * 3 * 2 + 5] = i * 4 + 3;
   }
-  geometry.setIndex(indices);
-  return geometry;
-}
+  geometry.setIndex(new THREE.BufferAttribute(indices, 1));
 
-const ZEROS = [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0];
+  geometry.drawRange.count = 0;
```

Comment:
Oh wow, we weren't setting this before? Very wasteful.

Reply:
The current algorithm is more complex than it needs to be. The sprites array and the attribute arrays are sparse. So I guess `geometry.drawRange.count` can't be used (efficiently). With the new algorithm suggested in this PR is easier and the arrays will be contiguous so that we can set `geometry.drawRange.count`. A beauty of simplicity.